src/echo_handler.h

```diff
@@ -10,6 +10,10 @@ class EchoHandler : public RequestHandler {
                 Response* response);
 
         virtual ~EchoHandler() = default;
+
+        virtual std::string type() const {
```

Comment:
I needed to be able to ask a handler what it's type was, or else the status handler would have no way of showing what type of handler is registered to a certain uri prefix. This seemed like a sane way to extend the API, as it is easy to define for new handler instances and yet gives straight-forward access to aid in logging, etc.